[auditbeat][8.4.0] Auditbeat frequently crashing, providing bad data, after upgrade from 8.3.3 to 8.4.0

After upgrading Auditbeat to 8.4.0 we noticed that it is frequently crashing with an error similar to the below on many of our hosts.
```
panic: runtime error: slice bounds out of range [:48] with length 32

goroutine 277 [running]:
github.com/elastic/go-libaudit/v2/auparse.parseSockaddr({0xc00083a537, 0x20})
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/auparse/sockaddr.go:65 +0xcc9
github.com/elastic/go-libaudit/v2/auparse.fieldMap.saddr(0xc000d6f740)
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/auparse/auparse.go:465 +0x159
github.com/elastic/go-libaudit/v2/auparse.(*AuditMessage).enrichData(0xc0001e96c0, 0xc000d6f740)
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/auparse/auparse.go:374 +0x23d
github.com/elastic/go-libaudit/v2/auparse.(*AuditMessage).Data(0xc0001e96c0)
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/auparse/auparse.go:116 +0x2ea
github.com/elastic/go-libaudit/v2/aucoalesce.addSockaddrRecord(0xc0001e96c0, 0xc000b59600)
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/aucoalesce/coalesce.go:321 +0x58
github.com/elastic/go-libaudit/v2/aucoalesce.normalizeCompound({0xc0005b53a0, 0x3, 0x4})
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/aucoalesce/coalesce.go:235 +0x2c6
github.com/elastic/go-libaudit/v2/aucoalesce.CoalesceMessages({0xc0005b53a0, 0x3, 0x4})
	/go/pkg/mod/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/aucoalesce/coalesce.go:184 +0x185
github.com/elastic/beats/v7/auditbeat/module/auditd.buildMetricbeatEvent({_, _, _}, {0x1, {0x55cf8d6da6aa, 0x6}, 0x2000, 0x0, 0x0, 0x0, ...})
	/go/src/github.com/elastic/beats/auditbeat/module/auditd/audit_linux.go:528 +0xa5
github.com/elastic/beats/v7/auditbeat/module/auditd.(*MetricSet).Run.func2()
	/go/src/github.com/elastic/beats/auditbeat/module/auditd/audit_linux.go:260 +0x29c
created by github.com/elastic/beats/v7/auditbeat/module/auditd.(*MetricSet).Run
	/go/src/github.com/elastic/beats/auditbeat/module/auditd/audit_linux.go:253 +0xd4e
```

Additionally, we have been observing this message in the logs since the upgrade:
```
"Unsuccessful conversion to generic event: 1 errors: [key=auditd.paths.1: error converting map[string]string to mapstr.M: error unmarshalling from JSON: invalid character 's' after object key:value pair]
```

Config:
```
auditbeat.modules:
- module: auditd
  audit_rules: |
    -a always,exit -F arch=b64 -S execve -S execveat -S accept -S accept4 -S connect -S bind -S listen
    -a always,exit -F dir=/var/log/audit/ -F perm=r -F auid>=500 -F auid!=unset
    -a always,exit -F dir=/var/log/audit/ -F perm=wa
- module: file_integrity
  paths:
  - /home/.ssh/*
  - /home/.aws*
  - /home/*
  - /root
  - /usr/sbin
  - /usr/bin
  - /sbin
  - /bin
  - /lib
  - /var/lib/docker
  - /etc/audit
  - /etc/libaudit.conf
  - /etc/cron.allow
  - /etc/cron.deny
  - /etc/cron.d/
  - /etc/cron.daily/
  - /etc/cron.hourly/
  - /etc/cron.monthly/
  - /etc/cron.weekly/
  - /etc/crontab
  - /etc/group
  - /etc/sudoers
  - /etc/hosts
  - /etc/network/
  - /etc/issue
  - /etc/motd
  - /etc/inittab
  - /etc/init.d/
  - /etc/init/
  - /etc/security/
  - /etc/ssh/sshd_config
  - /etc/systemd/
  - /etc/docker/daemon.json
  - /etc/modules*
  exclude_files:
  - '(?i)\.sw[nop]$'
  - '~$'
  - '/\.git($|/)'
  scan_at_start: true
  scan_rate_per_sec: 50 MiB
  max_file_size: 100 MiB
  hash_types: [sha256]
  recursive: true
- module: system
  datasets:
    - package
    - host
    - user
  period: 1m
  state.period: 24h
  user.detect_password_changes: true

processors:
- add_cloud_metadata:
- add_host_metadata:
- add_process_metadata:
    match_pids: [process.ppid]
    target: process.parent
```

For confirmed bugs, please report:
- Version: 8.4.0
- Operating System: official docker image
- Discuss Forum URL: N/A
- Steps to Reproduce:

--- a/CHANGELOG.next.asciidoc
+++ b/CHANGELOG.next.asciidoc
@@ -52,6 +52,7 @@ https://github.com/elastic/beats/compare/v8.2.0\...main[Check the HEAD diff]
 - auditd module: Fix minimum AuditStatus length so that library can support kernels from 2.6.32. {pull}32421[32421]
 - system/socket: Reduce memory usage of the dataset. {issue}32191[32191] {pull}32192[32192]
 - Fix rendering of MAC addresses to conform to ECS. {issue}32621[32621] {pull}32622[32622]
+- Fixes a bug with the auditd module where data is corrupted because it was not copied before the byte slice was reused. {issue}32818[32818] {pull}32823[32823]
 
 *Filebeat*
 

--- a/NOTICE.txt
+++ b/NOTICE.txt
@@ -10805,11 +10805,11 @@ Contents of probable licence file $GOMODCACHE/github.com/elastic/go-elasticsearc
 
 --------------------------------------------------------------------------------
 Dependency : github.com/elastic/go-libaudit/v2
-Version: v2.3.2-0.20220729123722-f8f7d5c19e6b
+Version: v2.3.2
 Licence type (autodetected): Apache-2.0
 --------------------------------------------------------------------------------
 
-Contents of probable licence file $GOMODCACHE/github.com/elastic/go-libaudit/v2@v2.3.2-0.20220729123722-f8f7d5c19e6b/LICENSE.txt:
+Contents of probable licence file $GOMODCACHE/github.com/elastic/go-libaudit/v2@v2.3.2/LICENSE.txt:
 
 
                                  Apache License

--- a/go.mod
+++ b/go.mod
@@ -75,7 +75,7 @@ require (
 	github.com/eclipse/paho.mqtt.golang v1.3.5
 	github.com/elastic/elastic-agent-client/v7 v7.0.0-20210727140539-f0905d9377f6
 	github.com/elastic/go-concert v0.2.0
-	github.com/elastic/go-libaudit/v2 v2.3.2-0.20220729123722-f8f7d5c19e6b
+	github.com/elastic/go-libaudit/v2 v2.3.2
 	github.com/elastic/go-licenser v0.4.0
 	github.com/elastic/go-lookslike v0.3.0
 	github.com/elastic/go-lumber v0.1.2-0.20220819171948-335fde24ea0f

